executor/analyze.go

```diff
@@ -115,6 +116,14 @@ func (e *AnalyzeExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 	for i := 0; i < concurrency; i++ {
 		e.wg.Run(func() { e.analyzeWorker(taskCh, resultsCh) })
 	}
+	pruneMode := variable.PartitionPruneMode(sessionVars.PartitionPruneMode.Load())
+	// needGlobalStats used to indicate whether we should merge the partition-level stats to global-level stats.
+	needGlobalStats := pruneMode == variable.Dynamic
+	globalStatsMap := make(map[globalStatsKey]globalStatsInfo)
+	g, _ := errgroup.WithContext(context.Background())
```

Comment:
Should we use the ctx passed from the `Next()`?

Reply:
Done